jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketDiagnosticsCollector.java

```diff
@@ -202,6 +204,42 @@ private void uriMismatchWarningCheck(IType type, List<Diagnostic> diagnostics, I
         }
     }
 
+    /*
+     * Create a warning diagnostic if a ServerEndpoint annotation does not follow the rules.
+     */
+    private void serverEndpointWarningCheck(IType type, List<Diagnostic> diagnostics, ICompilationUnit unit) throws JavaModelException {
+        for (IAnnotation annotation : type.getAnnotations()) {
+            if (annotation.getElementName().equals(WebSocketConstants.SERVER_ENDPOINT_ANNOTATION)) {
+                String path = annotation
+                        .getMemberValuePairs()[0]
```

Comment:
Are you sure that the URI will always be the first memberValuePair present for the Server Endpoint annotation? The annotation has other possible attributes such as encoders/decoders, so if those are specified first this will cause some unexpected errors. Probably best to iterate through these pairs and explicitly check that the key for the memberValuePair is 'value'.

See https://jakarta.ee/specifications/websocket/2.0/websocket-spec-2.0.html#serverendpoint for details.

Reply:
I'll look into it.